src/query/stream/streams.cpp

```diff
@@ -157,37 +158,238 @@ void from_json(const nlohmann::json &data, StreamStatus<TStream> &status) {
+      auto *arg_stream_name = procedure::Call<mgp_value *>(mgp_list_at, args, 0);
+      const auto *stream_name = procedure::Call<const char *>(mgp_value_get_string, arg_stream_name);
+      auto lock_ptr = streams_.Lock();
+      auto it = GetStream(*lock_ptr, std::string(stream_name));
+      std::visit(
+          utils::Overloaded{
+              [&](StreamData<KafkaStream> &kafka_stream) {
+                auto stream_source_ptr = kafka_stream.stream_source->Lock();
+                const auto info = stream_source_ptr->Info(kafka_stream.transformation_name);
+                mgp_result_record *record{nullptr};
+                {
+                  const auto success = TryOrSetError([&] { return mgp_result_new_record(result, &record); }, result);
+                  if (!success) {
+                    return;
+                  }
+                }
+
+                const auto consumer_group_value = GetStringValueOrSetError(info.consumer_group.c_str(), memory, result);
+                if (!consumer_group_value) {
+                  return;
+                }
+
+                mgp_list *topic_names{nullptr};
+                {
+                  const auto success = TryOrSetError(
+                      [&] { return mgp_list_make_empty(info.topics.size(), memory, &topic_names); }, result);
+                  if (!success) {
+                    return;
+                  }
+                }
+
+                for (const auto &topic : info.topics) {
+                  auto topic_value = GetStringValueOrSetError(topic.c_str(), memory, result);
+                  if (!topic_value) {
+                    return;
+                  }
+                  topic_names->elems.push_back(std::move(*topic_value));
+                }
+
+                procedure::MgpUniquePtr<mgp_value> topics_value{nullptr, mgp_value_destroy};
+                {
+                  const auto success = TryOrSetError(
+                      [&] { return procedure::CreateMgpObject(topics_value, mgp_value_make_list, topic_names); },
+                      result);
+                  if (!success) {
+                    return;
+                  }
+                }
+
+                const auto bootstrap_servers_value =
+                    GetStringValueOrSetError(info.bootstrap_servers.c_str(), memory, result);
+                if (!bootstrap_servers_value) {
+                  return;
+                }
+                const auto err1 = mgp_result_record_insert(record, "consumer_group", consumer_group_value.get());
+                const auto err2 = mgp_result_record_insert(record, "topics", topics_value.get());
+                const auto err3 = mgp_result_record_insert(record, "bootstrap_servers", bootstrap_servers_value.get());
+
+                if (err1 != MGP_ERROR_NO_ERROR || err2 != MGP_ERROR_NO_ERROR || err3 != MGP_ERROR_NO_ERROR) {
+                  static_cast<void>(mgp_result_set_error_msg(result, "Unable to set the result!"));
```

Comment:
Why we don't want to differentiate which one caused the error?

Reply:
not sure if it's necessary, maybe it could make sense from debugging perspective?